src/lib.rs

```diff
@@ -337,6 +337,15 @@ impl PosixACL {
         }
         Ok(())
     }
+
+    /// Returns a raw `acl_t` type that can be used directly in FFI calls to the acl library.
+    ///
+    /// The caller *must not*:
+    /// - `acl_free` or otherwise invalidate the returned `acl_t`
+    /// - use this `acl_t` after the PosixACL has been dropped
+    pub fn as_raw(&self) -> acl_sys::acl_t {
```

Comment:
Hmm, can you annotate the returned `acl_t` with the lifetime of `self`, so the compiler can verify the condition "_use this `acl_t` after the PosixACL has been dropped_"

If yes, I don't see any issue with omitting `unsfe`.

Reply:
I was thinking about this and I don't see any way to do it - ultimately, you're returning a pointer (as that's all an acl_t is).

I've ended up actually convincing myself that `fn into_raw` and `unsafe fn from_raw` are the right way to go - it's a well understood pattern in Rust (present in the stdlib) and hints at the pointer ownership issues you need to keep in mind.

(tt also means that you're not locked into an internal acl_t representation forever - you just need to be able to create from from your internal representation)

Reply:
Ah my bad, only `&` refs can have lifetimes, not `*` raw pointers.